src/client.rs

```diff
@@ -265,19 +271,36 @@ where
         &mut self,
         games: Option<Vec<String>>,
         slots: Option<Vec<String>>,
-        tags: Option<Vec<String>>,
+        tags: Vec<String>,
         data: serde_json::Value,
     ) -> Result<(), ArchipelagoError> {
         Ok(self
             .send(ClientMessage::Bounce(Bounce {
                 games,
                 slots,
                 tags,
-                data,
+                data: BounceData::Generic(data),
             }))
             .await?)
     }
 
+    /// Sends a death link notification to the server.
+    pub async fn death_link(&mut self, options: DeathLinkOptions) -> Result<(), ArchipelagoError> {
+        self.send(ClientMessage::Bounce(Bounce {
+            games: options.games,
+            slots: options.slots,
+            tags: vec![],
```

Comment:
```suggestion
            tags: vec!["DeathLink".to_string()],
```

DeathLink messages need this tag 
https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/network%20protocol.md#deathlink